dkpro-core-io-brat-asl/src/main/java/de/tudarmstadt/ukp/dkpro/core/io/brat/internal/model/BratTextAnnotation.java

```diff
@@ -29,84 +29,116 @@
     {
         this("T" + aId, aType, aBegin, aEnd, aText);
     }
 
-    public BratTextAnnotation(String aId, String aType, int aBegin, int aEnd, String aText)
+    public BratTextAnnotation(String aId, String aType, int[] aBegin, int[] aEnd, String[] aText)
     {
         super(aId, aType);
         begin = aBegin;
         end = aEnd;
         text = aText;
     }
 
-    public int getBegin()
+    public BratTextAnnotation(String aId, String aType, String aOffset, String aText)
+    {
+        super(aId, aType);
+        BratTextOffset bratTextOffset = new BratTextOffset(aOffset);
+        begin = bratTextOffset.getBegin();
+        end = bratTextOffset.getEnd();
+        text = splitText(aText, begin, end);
+     }
+      
+    private String[] splitText(String aText, int[] aBegin, int[] aEnd)
+    {
+        String[] result = new String[aBegin.length];
+        String pieceOfText = aText;
+        for (int i = 0; i < aBegin.length; i++) {
+            int size = aEnd[i] - aBegin[i];
+            result[i] = aText.substring(0, size);
+            pieceOfText = pieceOfText.substring(size);
+        }
+        return result;
+    }
+
+    public int[] getBegin()
     {
         return begin;
     }
 
-    public int getEnd()
+    public int[] getEnd()
     {
         return end;
     }
     
-    public String getText()
+    public String[] getText()
     {
         return text;
     }
     
     @Override
-    public void write(JsonGenerator aJG)
-        throws IOException
+    public void write(JsonGenerator aJG) throws IOException
     {
         // Format: [${ID}, ${TYPE}, [[${START}, ${END}]]]
         // note that range of the offsets are [${START},${END})
         // ['T1', 'Person', [[0, 11]]]
-        
+
         aJG.writeStartArray();
         aJG.writeString(getId());
         aJG.writeString(getType());
         aJG.writeStartArray();
-        aJG.writeStartArray();
-        aJG.writeNumber(begin);
-        aJG.writeNumber(end);
-        aJG.writeEndArray();
+        for (int i = 0; i < begin.length; i++) {
+            // handle discontinuous annotations
+            aJG.writeStartArray();
+            aJG.writeNumber(begin[i]);
+            aJG.writeNumber(end[i]);
+            aJG.writeEndArray();
+        }
         aJG.writeEndArray();
         aJG.writeEndArray();
     }
-    
+
     @Override
     public String toString()
     {
-        return getId() + '\t' + getType() + ' ' + begin + ' ' + end + '\t' + text;
+        return getId() + '\t' + getType() + ' ' + generateOffset(begin, end) + '\t' + String.join(" ", text);
     }
+    
+    private String generateOffset(int[] begin, int[] end)
+    {
+        StringBuilder sb = new StringBuilder();
+        for (int i = 0; i < begin.length; i++) {
+            sb.append(String.format("%s %s", begin[i], end[i]));
+            sb.append(";");
+
+        }
+        return sb.toString().replaceFirst(";$", "");
```

Comment:
Instead of doing this replace, just wrap the `sb.append(";");` into a `if (sb.length() > 0) { ... }` or even a `if (i > 0) {... }`